pkg/scheduler/internal/cache/cache_test.go

```diff
@@ -1854,6 +1860,106 @@ func TestSchedulerCache_updateNodeInfoSnapshotList(t *testing.T) {
 	}
 }
 
+func TestSchedulerCache_ErrorCases(t *testing.T) {
```

Comment:
Alternatively, you can test this in the related function instead of pull the false cases out and test it here, e.g "forget unknown pod" can go to "TestForgetPod".  It might be better to org these testcases, but I am fine with this as well.

Reply:
It was my first intent but once my code added I felt like messing up with the existing scenarii so I change my mind and gathered all false cases into this function

Reply:
+1, this is hard to maintain in the future. I didn't see any precedents in kubernetes. Let's try to make the tests pure.

Reply:
Sure thing. I will come back to move those tests to their related function.

Reply:
@chendave, @kerthcet,  I tried to keep tests as pure as possible and avoid messing too much with their clarity. This is what I ended with 807b29ad6e9120cbca49ae864b74a4f8ae003f36. 

Below the new unit tests code coverage for the `pkg/scheduler/internal/cache`.
<img width="708" alt="Screenshot 2022-12-05 at 16 44 08" src="https://user-images.githubusercontent.com/9576370/205680414-2fe47e50-5dfe-4124-862f-151bd72978b6.png">

All pipelines have passed, remaining at your disposal if you have any remark :) 

Best regards,